swap_meet/vendor.py

```diff
@@ -1,2 +1,107 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else: 
+            return False
+        
+    def get_by_id(self, item_id):
+
+        for item in self.inventory:
+            if item.id == item_id:
+                return item 
+
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item) 
+            self.inventory.append(their_item)
+            other_vendor.inventory.remove(their_item)
+            other_vendor.inventory.append(my_item)
+            return True
+        
+        return False
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        elif self.inventory and other_vendor.inventory:
+            self.inventory.append(other_vendor.inventory[0])
+            other_vendor.inventory.append(self.inventory[0])
+            self.inventory.remove(self.inventory[0]) 
+            other_vendor.inventory.remove(other_vendor.inventory[0])
+            return True
+        
+    def get_by_category(self, category):
+        
+        list_of_items = []
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                list_of_items.append(item)
+
+        return list_of_items
+
+    def get_best_by_category(self, category):
+
+        best_item = None
+        best_condition = -1
+
+        if not self.inventory:
+            return best_item
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                if item.condition > best_condition:
+                    best_condition = item.condition
+                    best_item = item
+
+                    if best_condition == 5:
+                        return best_item
+        
+        return best_item
+
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+
+        my_item = self.get_best_by_category(their_priority)
+        their_item = other_vendor.get_best_by_category(my_priority)
+        swap = self.swap_items(other_vendor, my_item, their_item)
+
+        return swap
+
+    def get_newest(self):
```

Comment:
👍 Nice helper to support your `swap_by_newest` method. If we didn't have a way to get the newest item for each `Vendor` that would make `swap_by_newest` a lot more complicated!